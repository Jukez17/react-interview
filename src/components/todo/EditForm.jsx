import React, { useContext } from 'react';
import { DispatchContext } from '../../context/Context';
import UseInputState from '../../hooks/UseInputState';
import useStyles from '../../styles/EditFormStyle';
import { EDIT_TODO } from '../../constants/Actions';

function EditTodoForm({ id, task, toggleEditForm }) {
  const classes = useStyles();
  const dispatch = useContext(DispatchContext);
  const [value, handleChange, clearValue] = UseInputState(task);

  return (
    <form
      testid="form"
      onSubmit={e => {
        e.preventDefault();
        dispatch({ type: EDIT_TODO, id, task: value });
        toggleEditForm();
        clearValue();
      }}
      className={classes.EditTodoForm}
    >
      <input
        testid='formInput'
        autoFocus
        value={value}
        onChange={handleChange}
        onClick={e => e.stopPropagation()}
        className={classes.input}
      />
    </form>
  );
}

export default EditTodoForm;