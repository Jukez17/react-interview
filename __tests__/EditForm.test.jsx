import React from "react";
import { renderHook, act } from "@testing-library/react-hooks";
import { render, fireEvent, getAllByTestId } from "@testing-library/react";
import EditForm from "../src/components/todo/EditForm";
import UseInputState from "../src/hooks/UseInputState";

it("renders EditForm", () => {
  render(<EditForm />);
});

describe("Form", () => {
  it("Actions", () => {
    const { result } = renderHook(() => UseInputState());
    const { handleChange } = result.current;
    const preventDefault = jest.fn()
    const e = preventDefault
    const MockComponent = () => (
      <>
        {["Hello"].map((value) => [
          <form
            data-testid="form"
            onSubmit={e}
          >
            <input
              key={value}
              data-testid="formInput"
              value={value}
              onChange={handleChange}
            />
          </form>,
        ])}
      </>
    );

    const { container, getByTestId } = render(<MockComponent />);
    const form = getByTestId("form");
    const inputField = getByTestId("formInput");

    act(() => {
      fireEvent.click(inputField);
    });

    expect(container).toMatchSnapshot();
  });
});
