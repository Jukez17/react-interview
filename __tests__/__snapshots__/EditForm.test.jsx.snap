// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Actions 1`] = `
<div>
  <form
    data-testid="form"
  >
    <input
      data-testid="formInput"
      value="Hello"
    />
  </form>
</div>
`;
